backend/metadata_writer/src/metadata_writer.py

```diff
@@ -98,15 +97,24 @@ def get_object_store_provider(endpoint: str) -> bool:
     else:
         return 'minio'
 
-def argo_artifact_to_uri(artifact: dict) -> str:
+def argo_artifact_to_uri(artifact: dict, skip_key=False) -> str:
     # s3 here means s3 compatible object storage. not AWS S3.
     if 's3' in artifact:
         s3_artifact = artifact['s3']
-        return '{provider}://{bucket}/{key}'.format(
-            provider=get_object_store_provider(s3_artifact['endpoint']),
-            bucket=s3_artifact.get('bucket', ''),
-            key=s3_artifact.get('key', ''),
-        )
+        if (s3_artifact.keys() >= {'endpoint', 'bucket'}):
```

Comment:
This comparison looks fragile. Can we do this check in a more robust way?

Reply:
Hm, and why does it look fragile to you?
Looks clean and short check of multiple keys in dict to me comparing to checking for subset or making unnecessary loops.